src/board.cpp

```diff
@@ -0,0 +1,190 @@
+    return;
+  }
+
+  gui_->print_header("NUMBERS ROYALE");
+  // Print players in reverse, so player 1 is on the bottom
+  for (int i = num_players() - 1; i >= 0; --i) {
+    std::string playerText = players_[i]->name() + ": " +
+        std::to_string(players_[i]->get_score()) + " points";
+    gui_->print_item(playerText.c_str());
+    gui_->print_item("Moves:");
+    gui_->print_item(players_[i]->get_moves_string().c_str(), true);
+  }
+  gui_->print_border();
+}
+
+void Board::print_scores() {
+  if (gui_ == nullptr) {
+    std::cerr << "No GUI object" << std::endl;
+    return;
+  }
+
+  int winner_id = check_game_winner();
+  switch (winner_id) {
+    case 0:
+      gui_->print_header("TIE GAME!");
+      break;
+    case -1:
+      gui_->print_header("SCOREBOARD", false);
+      break;
+    default:
+      std::string winner_text = players_[winner_id - 1]->name() + " WINS!";
+      gui_->print_header(winner_text.c_str());
+      break;
+  }
+  // Sort players by descending score
+  std::vector<Player*> sorted_players = players_;
+  std::sort(sorted_players.begin(), sorted_players.end(), score_cmp);
+
+  for (int i = 0; i < num_players(); ++i) {
+    std::string player_score_text = sorted_players[i]->name() + ": " +
+        std::to_string(sorted_players[i]->get_score()) + " points";
+    gui_->print_item(player_score_text.c_str());
+  }
+  gui_->print_border();
+}
+
+void Board::print_round_result() {
+  if (gui_ == nullptr) {
+    std::cerr << "No GUI object" << std::endl;
+    return;
+  }
+
+  int winner_id = check_round_winner(); 
+
+  // If game is over, print final score and exit
+  if (check_game_winner() != -1) {
+    print_scores();
+    return;
+  }
+
+  if (winner_id == 0) {
+    gui_->print_header("It's a tie!");
+  } else {
+    std::string winner_text = players_[winner_id - 1]->name() + " wins the round!";
+    gui_->print_header(winner_text.c_str());
+  }
+
+  // Sort players by last move, descending
+  std::vector<Player*> sorted_players = players_;
+  std::sort(sorted_players.begin(), sorted_players.end(), move_cmp);
+
+  for (int i = 0; i < num_players(); ++i) {
+    std::string player_move_text = sorted_players[i]->name() + " move: " +
+        std::to_string(sorted_players[i]->get_last_move());
+    gui_->print_item(player_move_text.c_str());
+  }
+
+  // Print the current scores
+  print_scores();
+}
+
+void Board::print_player_moves(unsigned int player_id) {
+  if (gui_ == nullptr) {
+    std::cerr << "No GUI object" << std::endl;
+    return;
+  }
+
+  gui_->print_header("NUMBERS ROYALE");
+  Player* player = players_[player_id - 1];
+  std::string player_move_text = player->name() + " moves:";
+  gui_->print_item(player_move_text.c_str());
+  gui_->print_item(player->get_moves_string().c_str(), true);
+  gui_->print_border();
+}
+
+void Board::add_player(bool is_cpu) {
+  if (is_cpu) {
+    players_.push_back(new CPU());
+  } else {
+    players_.push_back(new Player());
```

Comment:
e w.

Reply:
```suggestion
    players_.push_back(std::make_unique<CPU>());
  } else {
    players_.push_back(std::make_unique<Player>());
```
not sure if you need to do an additional std::move but idts